wimm.Missionary/FromStringConversion.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Reflection;
+
+namespace wimm.Missionary
+{
+    /// <summary>
+    /// A string conversion that's implemented by using either a constructor that accepts a single
+    /// string argument, or a public static Parse method to obtain the converted value. A type must
+    /// have at least of these methods available to be supported.
+    /// </summary>
+    /// <inheritDoc />
+    public class FromStringConversion<T> : IConversion<string, T>
+    {
+        private readonly DelegateConversion<string, T> _conversion;
+
+        /// <summary>
+        /// Indicates whether or not a type is supported.
+        /// </summary>
+        /// <typeparam name="T">The type.</typeparam>
+        /// <returns>
+        /// <c>true</c> is <typeparamref name="T"/> is supported, otherwise <c>false</c>.
+        /// </returns>
+        public static bool Supported()
+        {
+            return GetConversion() != null;
+        }
+
+        /// <summary>
+        /// Initializes a new new <see cref="FromStringConversion{T}"/>.
+        /// </summary>
+        /// <exception cref="NotSupportedException">
+        /// <typeparamref name="T"/> has neither a constructor that accepts a single string
+        /// argument, nor a public static method called Parse that returns an instance of
+        /// <typeparamref name="T"/>.
+        /// </exception>
+        public FromStringConversion()
+        {
+            _conversion = new DelegateConversion<string, T>(GetConversion() ??
+                throw new NotSupportedException($"{nameof(T)} does not support automatic conversion from string."));
+        }
+
+        public T Convert(string from) => _conversion.Convert(from);
+
+        private static Func<string, T> GetConversion()
+        {
+            var constructor = typeof(T).GetConstructor(new[] { typeof(string) });
+            if (constructor != null)
+                return str => (T)constructor.Invoke(new[] { str });
+
+            var name = "Parse";
+            var bindingFlags = BindingFlags.Static | BindingFlags.Public;
+            var types = new[] { typeof(string) };
+            var parseMethod = typeof(T).GetMethod(name, bindingFlags, null, types, null);
+            if (parseMethod != null && parseMethod.ReturnType == typeof(T))
+                return str => (T)parseMethod.Invoke(null, new[] { str });
+
+            return null;
```

Comment:
![image](https://user-images.githubusercontent.com/7606469/46265281-59f71300-c4fb-11e8-94de-de0776fbebb0.png)

Reply:
Once it's feature complete for 1.0 I'll publish Secundatives and Maybeify this whole thing.

Reply:
bump